aten/src/ATen/core/ivalue_inl.h

```diff
@@ -561,37 +548,29 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
       const at::IValue& value,
       c10::optional<std::vector<std::reference_wrapper<const at::DataPtr>>>
           dataPtrs) {
-    if (impl_ == nullptr) {
```

Comment:
Yes indeed. I admit I didn't benchmark as I assumed that ifs and empty loops would be negligible. Do you know some way I could measure this to make sure?

Reply:
I agree that most likely it won't lead to any noticeable differences in user land. So please feel free to skip measurements. The easiest way to measure would be creating a bunch of `torch.futures.Future` and mark them as completed. Though there are layers on top of the raw Future impl, but I think it should still be fine since this is how users gonna use it.